### PR TITLE
feat(developer): add CompilerMessages support to kmc-package

### PR DESCRIPTION
**Pull request by mcdurdin**
Relates to #8150.

Adds basic infrastructure for CompilerMessages events, along with file-existence check and absolute path warnings. Additional warnings,
errors, and hints to come.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-package/src/messages.ts
Comment (mcdurdin) [inline developer/src/kmc-package/src/messages.ts:1]: This file matches kmc-keyboard messages.ts format.

File: developer/src/kmc-package/test/test-package-compiler.ts [hunk 1]
```diff
@@ -117,4 +123,27 @@ describe('KmpCompiler', function () {
...
+
+    assert.doesNotThrow(() => {
+      kmpJson = kmpCompiler.transformKpsToKmpObject(source, kpsPath);
+    });
+
+    await assert.isNull(kmpCompiler.buildKmpFile(kpsPath, kmpJson));
+
+    assert.lengthOf(callbacks.messages, 2);
+    assert.deepEqual(callbacks.messages[0].code, CompilerMessages.WARN_AbsolutePath);
+    assert.deepEqual(callbacks.messages[1].code, CompilerMessages.ERROR_FileDoesNotExist); //TODO: this should be a file-missing-error
```
Comment (mcdurdin) [inline developer/src/kmc-package/test/test-package-compiler.ts:146]: comment already fixed on subsequent PR #8628

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.91-alpha](https://keyman.com/downloads/releases/alpha/17.0.91)

**Pull request merged**